File: 2493-reverse-odd-levels-of-binary-tree/2493-reverse-odd-levels-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void inOrder(TreeNode node1, TreeNode node2, int level)
    {
        if(node1 == null || node2 == null)
        {
            return;
        }
        if(level % 2 != 0)
        {
            int temp = node1.val;
            node1.val = node2.val;
            node2.val = temp;
        }
        inOrder(node1.left, node2.right, level + 1);
        inOrder(node2.left, node1.right, level + 1);
    }
    public TreeNode reverseOddLevels(TreeNode root) {
        inOrder(root.left, root.right, 1);
        return root;
    }
}

File: 2493-reverse-odd-levels-of-binary-tree/2493-reverse-odd-levels-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def reverseOddLevels(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def inOrder(node1, node2, level):
            if node1 is None or node2 is None:
                return
            if(level % 2 != 0):
                node1.val, node2.val = node2.val, node1.val
            inOrder(node1.left, node2.right, level + 1)
            inOrder(node2.left, node1.right, level + 1)
            
        inOrder(root.left, root.right, 1)
        return root